test/test_dataflow.py

```diff
@@ -1,51 +1,78 @@
 from unittest import TestCase
+from unittest.mock import patch
+
 import apache_beam as beam
 from apache_beam import Pipeline
 from apache_beam.options.pipeline_options import PipelineOptions
-from apache_beam.testing.util import assert_that, equal_to
-
+from apache_beam.runners.portability.fn_api_runner.fn_runner import RunnerResult
+from apache_beam.testing.test_pipeline import TestPipeline
+from collections import defaultdict
 from bigflow import JobContext, Workflow
 
 from bigflow.dataflow import BeamJob
 
 
 class CountWordsFn(beam.DoFn):
+    def __init__(self, save):
+        super().__init__()
+        self.save = save
+
     def process(self, element, *args, **kwargs):
         word, count = element
+        self.save.counter[word] += len(count)
         yield word, len(count)
 
 
+class Save(object):
```

Comment:
Why not to use just `defaultdict(int)` instead of this `wrapping` class?

Reply:
In that case `self.assertEqual(driver.saver.save.counter, {'valid': 2, 'word': 1})` is never True. driver.saver.save.counter has no values,

Reply:
Well, looks like it is effectively a global variable (aka 'static' field).
So you could replace `Save.counter` with something like `GLOBAL_COUNTER`, which is not the best approach, but a little more readable.

Reply:
I will stay with Save class

Reply:
Well, maybe at least provide more verbose name/comment, because it is quite hard to understand what is `Save`